velox/type/Conversions.h

```diff
@@ -28,159 +28,214 @@ DECLARE_bool(experimental_enable_legacy_cast);
+  bool legacyCast = false;
 };
 
-template <bool LEGACY_CAST>
-struct Converter<TypeKind::BOOLEAN, void, false, LEGACY_CAST> {
+template <>
+struct Converter<TypeKind::BOOLEAN> {
   using T = bool;
 
   template <typename From>
-  static T cast(const From& v) {
+  T cast(const From& v) {
+    if (truncate) {
+      VELOX_UNSUPPORTED("Conversion to BOOLEAN is not supported");
+    }
     return folly::to<T>(v);
   }
 
-  static T cast(folly::StringPiece v) {
+  T cast(folly::StringPiece v) {
     return folly::to<T>(v);
   }
 
-  static T cast(const StringView& v) {
+  T cast(const StringView& v) {
     return folly::to<T>(folly::StringPiece(v));
   }
 
-  static T cast(const std::string& v) {
+  T cast(const std::string& v) {
+    return folly::to<T>(v);
+  }
+
+  T cast(const bool& v) {
     return folly::to<T>(v);
   }
 
-  static T cast(const Timestamp&) {
+  T cast(const float& v) {
+    if (truncate) {
+      if (std::isnan(v)) {
+        return false;
+      }
+      return v != 0;
+    } else {
+      return folly::to<T>(v);
+    }
+  }
+
+  T cast(const double& v) {
+    if (truncate) {
+      if (std::isnan(v)) {
+        return false;
+      }
+      return v != 0;
+    } else {
+      return folly::to<T>(v);
+    }
+  }
+
+  T cast(const int8_t& v) {
+    if (truncate) {
+      return T(v);
+    } else {
+      return folly::to<T>(v);
+    }
+  }
+
+  T cast(const int16_t& v) {
+    if (truncate) {
+      return T(v);
+    } else {
+      return folly::to<T>(v);
+    }
+  }
+
+  T cast(const int32_t& v) {
+    if (truncate) {
+      return T(v);
+    } else {
+      return folly::to<T>(v);
+    }
+  }
+
+  T cast(const int64_t& v) {
+    if (truncate) {
+      return T(v);
+    } else {
+      return folly::to<T>(v);
+    }
+  }
+
+  T cast(const Timestamp&) {
     VELOX_UNSUPPORTED("Conversion of Timestamp to Boolean is not supported");
   }
+
+  bool truncate = false;
+  bool legacyCast = false;
 };
 
-template <TypeKind KIND, bool TRUNCATE, bool LEGACY_CAST>
+template <TypeKind KIND>
 struct Converter<
     KIND,
     std::enable_if_t<
-        KIND == TypeKind::BOOLEAN || KIND == TypeKind::TINYINT ||
-            KIND == TypeKind::SMALLINT || KIND == TypeKind::INTEGER ||
-            KIND == TypeKind::BIGINT || KIND == TypeKind::HUGEINT,
-        void>,
-    TRUNCATE,
-    LEGACY_CAST> {
+        KIND == TypeKind::TINYINT || KIND == TypeKind::SMALLINT ||
+            KIND == TypeKind::INTEGER || KIND == TypeKind::BIGINT ||
+            KIND == TypeKind::HUGEINT,
+        void>> {
   using T = typename TypeTraits<KIND>::NativeType;
 
   template <typename From>
-  static T cast(const From&) {
+  T cast(const From&) {
     VELOX_UNSUPPORTED(
         "Conversion to {} is not supported", TypeTraits<KIND>::name);
   }
 
-  static T convertStringToInt(const folly::StringPiece v) {
-    // Handling boolean target case fist because it is in this scope
-    if constexpr (std::is_same_v<T, bool>) {
-      return folly::to<T>(v);
```

Comment:
I didn't make much change to this static helper func. Only removed the if-condition
```
if constexpr (std::is_same_v<T, bool>) {
      return folly::to<T>(v);
```
GitHub is not telling the difference very well